tests/ServiceLayer.Mesh.Tests/Functions/FileRetryFunctionTests.cs

```diff
@@ -0,0 +1,194 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Moq;
+using NHS.MESH.Client.Contracts.Services;
+using ServiceLayer.Mesh.Data;
+using ServiceLayer.Mesh.Functions;
+using ServiceLayer.Mesh.Messaging;
+using ServiceLayer.Mesh.Models;
+using ServiceLayer.Mesh.Configuration;
+
+namespace ServiceLayer.Mesh.Tests.Functions;
+
+public class FileRetryFunctionTests
+{
+    private readonly Mock<ILogger<FileRetryFunction>> _loggerMock;
+    private readonly Mock<IMeshInboxService> _meshInboxServiceMock;
+    private readonly Mock<IFileExtractQueueClient> _fileExtractQueueClientMock;
+    private readonly Mock<IFileTransformQueueClient> _fileTransformQueueClientMock;
+    private readonly Mock<IFileRetryFunctionConfiguration> _configuration;
+    private readonly ServiceLayerDbContext _dbContext;
+    private readonly FileRetryFunction _function;
+
+    public FileRetryFunctionTests()
+    {
+        _loggerMock = new Mock<ILogger<FileRetryFunction>>();
+        _meshInboxServiceMock = new Mock<IMeshInboxService>();
+        _fileExtractQueueClientMock = new Mock<IFileExtractQueueClient>();
+        _fileTransformQueueClientMock = new Mock<IFileTransformQueueClient>();
+        _configuration = new Mock<IFileRetryFunctionConfiguration>();
+
+        var options = new DbContextOptionsBuilder<ServiceLayerDbContext>()
+            .UseInMemoryDatabase(databaseName: Guid.NewGuid().ToString())
+            .ConfigureWarnings(warnings =>
+                warnings.Ignore(Microsoft.EntityFrameworkCore.Diagnostics.InMemoryEventId.TransactionIgnoredWarning))
+            .Options;
+
+        _dbContext = new ServiceLayerDbContext(options);
+
+        _configuration.Setup(c => c.StaleHours).Returns(12);
+
+        _function = new FileRetryFunction(
+            _loggerMock.Object,
+            _dbContext,
+            _fileExtractQueueClientMock.Object,
+            _fileTransformQueueClientMock.Object,
+            _configuration.Object
+        );
+    }
+
+    [Theory]
+    [InlineData(MeshFileStatus.Discovered)]
+    [InlineData(MeshFileStatus.Extracting)]
+    [InlineData(MeshFileStatus.Extracted)]
+    [InlineData(MeshFileStatus.Transforming)]
+    public async Task Run_EnqueuesFileOlderThan12Hours(MeshFileStatus testStatus)
+    {
+        var file = new MeshFile
+        {
+            FileType = MeshFileType.NbssAppointmentEvents,
+            MailboxId = "test-mailbox",
+            FileId = "file-1",
+            Status = testStatus,
+            LastUpdatedUtc = DateTime.UtcNow.AddHours(-13)
+        };
+
+        _dbContext.MeshFiles.Add(file);
+        await _dbContext.SaveChangesAsync();
+
+        // Act
+        await _function.Run(null);
+
+        // Assert
+        if (testStatus == MeshFileStatus.Discovered || testStatus == MeshFileStatus.Extracting)
```

Comment:
avoid logic in the assertion section of the tests

split this into two tests - one for retrying extractions, and another for retrying transformations

Reply:
Also, having split it, add assertions that the queue client mock that doesn't expect to receive any calls isn't invoked...